gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -131,12 +131,17 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
 
         return api_endpoint.replace(".googleapis.com", ".mtls.googleapis.com")
 
+    # Note: DEFAULT_ENDPOINT is deprecated. Use DEFAULT_ENDPOINT_TEMPLATE instead.
     DEFAULT_ENDPOINT = {% if service.host %}"{{ service.host }}"{% else %}None{% endif %}
 
     DEFAULT_MTLS_ENDPOINT = _get_default_mtls_endpoint.__func__(  # type: ignore
         DEFAULT_ENDPOINT
     )
 
+    DEFAULT_ENDPOINT_TEMPLATE = {% if service.host %}"{{ service.host.replace("googleapis.com", "{UNIVERSE_DOMAIN}") }}"{% else %}None{% endif %}
+    
+    GOOGLE_DEFAULT_UNIVERSE = "googleapis.com"
```

Comment:
`GOOGLE_DEFAULT_UNIVERSE` can definitely be moved to `api_core` but not `DEFAULT_ENDOINT_TEMPLATE` since the value for it is dynamically resolved.

Note that we're using an older version of `api_core` for `python 3.7`. As a result we'll have to have the check `hasattr(Client_options, "universe_domain")` every time we use code from `api_core` in gapic that we recently add.

Reply:
So then we would not have custom universes for Python 3.7? Does that go against the requirements of this project, or is that acceptable?

If that would be a problem, you can keep the code as is but add a TODO outlining two options I see (do you see others?):
*  when we stop supporting Python 3.7, we can move this symbol to api_core; or
* we could add the symbol to the latest version of `api_core` and at the same time:
  *  take the version of `api_core` we support for 3.7, make a branch for that, and there only add what we need for this feature. But we'd have to be clear on how to release it and on making sure we pin the dependency correctly. 


(Probably not worth investing the time doing all that coding work at the moment, which is why I'm suggesting a TODO if we need to support the feature in 3.7)

Reply:
> So then we would not have custom universes for Python 3.7? Does that go against the requirements of this project, or is that acceptable?

The issue is not specific to python 3.7 but rather whether the user has an older version of `google-api-core` installed and wants to use the new feature which requires a newer version of `google-api-core`

Possible solution : If the user sets `universe_domain` and has an older version of `google-api-core` installed, we can raise an error that a newer version of `google-api-core`  is required (we should provide the users with specific minimum version).

Reply:
As per our off-line discussion, we'll defer refactoring some functionality into `google-api-core` until after this PR is in.

Reply:
And I withdraw my original suggestion here. The string substitution is happening at the Jinjha template level, so what I suggested does not make sense.

That said, I would strongly encourage taking this business logic out of the templates and into the generator, but that should probably be part of a larger refactoring.